tests/dsl/Priority.cpp

```diff
@@ -19,48 +19,64 @@
-            REQUIRE(!low);
-            REQUIRE(!med);
-            REQUIRE(!high);
-
-            high = true;
-        });
-
-        on<Trigger<Message2>, Priority::NORMAL>().then("Normal", [] {
-            // We should be the second to run
-            REQUIRE(!low);
-            REQUIRE(!med);
-            REQUIRE(high);
-
-            med = true;
-        });
-
-        on<Trigger<Message1>, Priority::LOW>().then("Low", [this] {
-            // We should be the final one to run
-            REQUIRE(!low);
-            REQUIRE(med);
-            REQUIRE(high);
-
-            low = true;
-
-            // We're done
-            powerplant.shutdown();
+    TestReactor(std::unique_ptr<NUClear::Environment> environment) : TestBase(std::move(environment)) {
+
+        // Declare in the order you'd expect them to fire
+        on<Trigger<Message<1>>, Priority::REALTIME>().then([] { events.push_back("Realtime Message<1>"); });
+        on<Trigger<Message<1>>, Priority::HIGH>().then("High", [] { events.push_back("High Message<1>"); });
+        on<Trigger<Message<1>>>().then([] { events.push_back("Default Message<1>"); });
+        on<Trigger<Message<1>>, Priority::NORMAL>().then("Normal", [] { events.push_back("Normal Message<1>"); });
+        on<Trigger<Message<1>>, Priority::LOW>().then("Low", [] { events.push_back("Low Message<1>"); });
+        on<Trigger<Message<1>>, Priority::IDLE>().then([] { events.push_back("Idle Message<1>"); });
+
+        // Declare in the opposite order to what you'd expect them to fire
+        on<Trigger<Message<2>>, Priority::IDLE>().then([] { events.push_back("Idle Message<2>"); });
+        on<Trigger<Message<2>>, Priority::LOW>().then([] { events.push_back("Low Message<2>"); });
+        on<Trigger<Message<2>>, Priority::NORMAL>().then([] { events.push_back("Normal Message<2>"); });
+        on<Trigger<Message<2>>>().then([] { events.push_back("Default Message<2>"); });
+        on<Trigger<Message<2>>, Priority::HIGH>().then([] { events.push_back("High Message<2>"); });
+        on<Trigger<Message<2>>, Priority::REALTIME>().then([] { events.push_back("Realtime Message<2>"); });
+
+        // Declare in a random order
+        std::array<int, 5> order = {0, 1, 2, 3, 4};
+        std::shuffle(order.begin(), order.end(), std::mt19937(std::random_device()()));
+        for (const auto& i : order) {
+            switch (i) {
+                case 0:
+                    on<Trigger<Message<3>>, Priority::REALTIME>().then([] { events.push_back("Realtime Message<3>"); });
+                    break;
+                case 1:
+                    on<Trigger<Message<3>>, Priority::HIGH>().then([] { events.push_back("High Message<3>"); });
+                    break;
+                case 2:
+                    on<Trigger<Message<3>>, Priority::NORMAL>().then([] { events.push_back("Normal Message<3>"); });
+                    on<Trigger<Message<3>>>().then([] { events.push_back("Default Message<3>"); });
```

Comment:
Random order has to be missing one as otherwise the Default and Normal (which are both normal priorities) would end up flipping randomly. Doesn't work so well when you have an expected list.